### PR TITLE
ref: only override SENTRY_NEWSLETTER under test if it hasn't already been overridden

### PR DESCRIPTION
**Pull request by asottile-sentry**
this fixes a (not running because misnamed) test in getsentry

### CODE REVIEW

File: requirements-dev-only-frozen.txt [hunk 1]
```diff
@@ -12,7 +12,7 @@ distlib==0.3.4
 docker==3.7.0
 docker-pycreds==0.4.0
 exam==0.5.1
-filelock==3.7.0
+filelock==3.7.1
```
Comment (joshuarli) [inline requirements-dev-only-frozen.txt:15]: What's this for?

Comment (joshuarli) [reply requirements-dev-only-frozen.txt:15]: Oh, I see. Assuming you undo this.

**Review by joshuarli — APPROVED** [review]
lgtm pending https://github.com/getsentry/sentry/pull/35434#discussion_r891770713

File: requirements-dev-only-frozen.txt (hunk 1 above)
Comment (asottile-sentry) [reply requirements-dev-only-frozen.txt:15]: it's only used in pre-commit (via `virtualenv`) so I'm just going to bump it anyway

**Pull request merged**